src/clj/puppetlabs/services/jruby/jruby_puppet_schemas.clj

```diff
@@ -183,3 +182,39 @@
   "Schema for a clojure persistent map for the system environment"
   (schema/both EnvMap
     (schema/either PersistentArrayMap PersistentHashMap)))
+
+(defn event-type-requested?
+  [e]
+  (= :instance-requested (:type e)))
+
+(defn event-type-borrowed?
+  [e]
+  (= :instance-borrowed (:type e)))
+
+(defn event-type-returned?
+  [e]
+  (= :instance-returned (:type e)))
+
+(def JRubyEventAction
```

Comment:
While looking over this PR it took me a bit to realize what `action` represented.  I thought it was going to be a function, considering the name, but then the `schema/Any` seemed a bit odd.

Turns out it represents a "reason", and it's generally a map (at least that's what the docstrings below say).

Since the data type can be anything, what are we thinking we would do with the data?  I see you mention logging and metrics, but if it can be anything what more can we do than call `str` on it?

So I guess my questions are:
- What about a different name, like `JRubyEventReason`?  The term "action" seems overloaded here
- Why is the datatype `schema/Any`?